src/clockcache.c

```diff
@@ -1873,33 +1877,41 @@ clockcache_init(clockcache        *cc,   // OUT
    return STATUS_NO_MEMORY;
 }
 
+/*
+ * De-init the resources allocated to initialize a clockcache.
+ * This function may be called to deal with error situations, or a failed
+ * clockcache_init(). So check for non-NULL handles before trying to release
+ * resources.
+ */
 void
 clockcache_deinit(clockcache *cc) // IN/OUT
 {
    platform_assert(cc != NULL);
 
-   /*
-    * Check for non-null cause this is also used to clean up a failed
-    * clockcache_init
-    */
    if (cc->logfile) {
       clockcache_log(0, 0, "deinit %s\n", "");
 #if defined(CC_LOG) || defined(ADDR_TRACING)
       platform_close_log_file(cc->logfile);
 #endif
    }
 
-   platform_buffer_deinit(&cc->rc_bh);
+   if (cc->lookup) {
+      platform_free(cc->heap_id, cc->lookup);
+   }
+   if (cc->entry) {
+      platform_free(cc->heap_id, cc->entry);
+   }
 
-   platform_free(cc->heap_id, cc->entry);
-   platform_free(cc->heap_id, cc->lookup);
    platform_buffer_deinit(&cc->bh);
+   platform_buffer_deinit(&cc->rc_bh);
```

Comment:
I did not bracket this with a corresponding `if (!NULL)` check on the member field as I am relying on the the `bh` struct being zero'ed out on declaration. 

Current code always conforms to this ... Let's talk this point through ...